src/main/java/com/faforever/client/theme/Theme.java

```diff
@@ -13,21 +15,20 @@
   private static final String AUTHOR = "author";
   private static final String COMPATIBILITY_VERSION = "compatibilityVersion";
   private static final String THEME_VERSION = "themeVersion";
+  private static final String NEEDS_RESTART = "needsRestart";
 
-  private StringProperty displayName;
-  private StringProperty author;
-  private SimpleObjectProperty<Integer> compatibilityVersion;
-  private StringProperty themeVersion;
+  private final StringProperty displayName;
+  private final StringProperty author;
+  private final SimpleObjectProperty<Integer> compatibilityVersion;
+  private final StringProperty themeVersion;
+  private final BooleanProperty needsRestart;
 
-  public Theme() {
-    this(null, null, null, null);
-  }
-
-  public Theme(String displayName, String author, Integer compatibilityVersion, String themeVersion) {
+  public Theme(String displayName, String author, Integer compatibilityVersion, String themeVersion, boolean needsRestart) {
     this.displayName = new SimpleStringProperty(displayName);
     this.author = new SimpleStringProperty(author);
     this.compatibilityVersion = new SimpleObjectProperty<>(compatibilityVersion);
     this.themeVersion = new SimpleStringProperty(themeVersion);
+    this.needsRestart = new SimpleBooleanProperty(needsRestart);
```

Comment:
I'm still against having such a property.

Reply:
So what are u for. No restart pop up, always a pop up? Looking if it contains fxml files?(that's inaccurate) Looking if it contains none css files?